src/database/schema/ChannelListSchema.ts

```diff
@@ -515,14 +516,17 @@ class ChannelList implements BaseDbSchema {
     });
   }
 
-  static fromInitAnonymousChatAPI(data: ModifyAnonymousChatData): ChannelList {
+  static fromInitAnonymousChatAPI(
+    data: ModifyAnonymousChatData,
+    type: 'ANON_PM' | 'PM'
+  ): ChannelList {
     return new ChannelList({
       id: data?.message?.cid,
-      channelPicture: '',
+      channelPicture: data?.targetImage || DEFAULT_PROFILE_PIC_PATH,
       name: data?.targetName,
       description: data?.message?.message,
       unreadCount: 0,
-      channelType: 'ANON_PM',
+      channelType: type,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made to the `fromInitAnonymousChatAPI` function are logical and well-implemented. The function now accepts an additional parameter `type` which is used to set the `channelType`. Also, the `channelPicture` is now set based on `data.targetImage` or falls back to `DEFAULT_PROFILE_PIC_PATH` if `data.targetImage` is falsy. This makes the function more flexible and robust.
```diff
-  static fromInitAnonymousChatAPI(data: ModifyAnonymousChatData): ChannelList {
+  static fromInitAnonymousChatAPI(
+    data: ModifyAnonymousChatData,
+    type: 'ANON_PM' | 'PM'
+  ): ChannelList {
     return new ChannelList({
       id: data?.message?.cid,
-      channelPicture: '',
+      channelPicture: data?.targetImage || DEFAULT_PROFILE_PIC_PATH,
       name: data?.targetName,
       description: data?.message?.message,
       unreadCount: 0,
-      channelType: 'ANON_PM',
+      channelType: type,
       lastUpdatedAt: data?.message?.created_at,
       lastUpdatedBy: data?.message?.user?.id,
       createdAt: data?.message?.created_at,
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->